src/ui-setup/scripts/generate-ui.js

```diff
@@ -1,34 +1,48 @@
-const ncp = require('ncp')
 const fs = require('fs')
+const fse = require('fs-extra')
 const chalk = require('chalk')
 const path = require('path')
+const { exec } = require('child_process')
 const renderSchema = require('./renderSchema')
 const renderResolvers = require('./renderResolvers')
 const renderHomePage = require('./renderHomePage')
 const renderTypePage = require('./renderTypePage')
+const renderTypePageTest = require('./renderTypePageTest')
+const renderTypePageIntegrationTest = require('./renderTypePageIntegrationTest')
 const renderIndex = require('./renderIndex')
 const mapObject = require('./render-utils').mapObject
 const typeSpecPath = process.argv[2]
-const defaultTypeSpecPath = path.resolve("src/setup/type-specs", "sample-type-spec.json");
+const defaultTypeSpecPath = path.resolve('src/setup/type-specs', 'sample-type-spec.json')
 const defaultTypeSpec = require(defaultTypeSpecPath)
+const { capitalize, toCamelCase } = require('../../setup/utils.js')
+
+// map over type spec to ensure the types are all capitalized before feeding the name to the UI generation
+const formCSTypeSpec = typeSpec => {
+  const rawTypeSpecTypesMap = mapObject(typeSpec.types, (typeName, type) => [capitalize(toCamelCase(typeName)), type])
```

Comment:
I would call this variable

```capitalizedTypeArray```

`raw` and `map` are both a little misleading in this context

Reply:
I think of an array of key/value pairs as a map, which is why I choose Map for the title - I'm fine with Array, instead though.  I agree that raw is unneeded. :+1: